lib/MetaCPAN/Script/Mapping.pm

```diff
@@ -156,8 +192,20 @@ sub delete_index {
     $self->_delete_index($name);
 }
 
+sub delete_all {
+    my $self = $_[0];
+    my $name = undef;
+
+    $self->are_you_sure("ALL Indices will be deleted !!!");
```

Comment:
this is why I think the --yes flag will at some point bite us.
what is the need for the bypass?

Reply:
Thank you @mickeyn  for taking your time to look into my work in depth.
The idea behind the `--yes` flag was basically the same as with `IO::Interactive::is_interactive()` only that the `--yes` flag would be operator introduced.
In the tests I had the impression that the `is_interactive()` function would not detect properly the docker environment but the test without the `--yes` option 
documented at:
[Indexation Test without `--yes` option ](https://github.com/bodo-hugo-barwich/metacpan-docker/runs/4486607335?check_suite_focus=true#step:8:24)
showed that the test environment was correctly recognized by the module.

So, I removed the `--yes` option and all the documentation about it.

The `--delete --all` combination is meant to clear a corrupted _ElasticSearch_ environment completely to enable a correct rebuild.
This also has to be handled with **caution** since any indices which are not from the _MetaCPAN_ project will also be deleted.

Reply:
If there's a need for non-production env. we can compromise by coupling it with some env flag.
I just don't want to allow this in production env.

Reply:
Actually, I don't have the background knowledge about the project as you do.
So, I don't understand quite well what you have in mind with "_env flag_".

The issue that this development wants to fix does not happen in the testing environment since those tests are always run on an **empty cluster**.
The **index corruption** is more likely to happen in the **installation phase** and then the feature `delete_all()` will be used by the operator to start the installation all over.
_ElasticSearch_ index corruption can also happen when the _ElasticSearch_ Engine crashes during the Indexation Sequence - perhaps as an out-of-memory crash in a Kubernetes Environment.
Although this fix is important for the Docker Deployment it is actually not a docker-only issue and thus not limited to a development or testing environment.
I think an Ansible Installation Automation could also run into this issue even on a Virtual or BareMetal Server.

Reply:
I mean that the site (our production index) cannot be put at risk of accidental destruction by having this flags combination in code that could potentially run on production servers.
(having worked on the ES version migration with a lot of manual CLI script executions, I can imagine a few scenarios where it could have happened and the interactive verification check was added for a reason and saved us a couple of times)

Reply:
Hi, @mickeyn . Perhaps my comment below was a bit too scarce that it was not quite clear but I excluded all the `--delete --all` and `--yes` logic from this branch and moved it to another branch.
I opened a new issue to discuss whether this logic is actually still needed. (see #1048)